CHANGELOG.md

```diff
@@ -6,13 +6,13 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic
 Versioning](http://semver.org/spec/v2.0.0.html).
 
-## [Unreleased]
+## [0.6.0] - 2024-??
 
 We recommend using the following SYCL versions with this release:
 
-- DPC++: ???
+- DPC++: 89327e0a or newer
 - AdaptiveCpp (formerly hipSYCL): v24.06
-- SimSYCL: ???
+- SimSYCL: master
 
 See our [platform support guide](docs/platform-support.md) for a complete list of all officially supported configurations.
 
```

Comment:
What do you think of adding an "Upgrade Guide" section in this location, with at least three entries:

* One that explains that run scripts on clusters need to be changed to only start one process per node but give it all the GPUs.
* The one about clean builds due to submodule dependency updates (removing the note below)
* Device selection

Basically a summary, in order of importance, of things that cause a notable user-facing change when upgrading existing projects.

Reply:
I've added such a section at the top of the changelog